src/entities/user/user.validation.ts

```diff
@@ -23,7 +23,7 @@ export const validateCreateUser = (
     login: LOGIN_RULES.required(),
     password: PASSWORD_RULES.required(),
     age: AGE_RULES.required(),
-    isDeleted: IS_DELETED_RULES,
+    is_deleted: IS_DELETED_RULES,
```

Comment:
I believe this should be applied to validateUpdateUser too.
Also consider using dataMapper design pattern for this - converting object commonly used in app to object used by DB. This pattern was demonstrated in learn videos for this module.

Reply:
I will allow `isDeleted` to be passed when updating the user (though it's weird). But I would not allow passing `isDeleted` when creating a new user (not that useful to create a soft-deleted user :smile:).

Reply:
I reverted `isDeleted` to be a "domain name" and `is_deleted` will be in the data layer (DB). I think in our services when we update the user and use Sequelize like this: 

```
    const { login, password, age, isDeleted } = userDto;
    const userToUpdate = await UserModel.findOne({
      where: {
        id: userId,
      },
    });

    if (userToUpdate) {
      if (isDeleted) userToUpdate.is_deleted = isDeleted;
      ...
   }
```
this already acts like a data mapper, since Typescript/Sequelize won't allow me to call .isDeleted on `userToUpdate` and `userToUpdate` is already a database entity that knows nothing about `isDeleted` (since it's domain-specific).

So I think we should not implement any additional logic here even if we keep both `isDeleted` and `is_deleted` together in our app.

Reply:
But: the thing I noticed is that after update when I want to send the user the updated entity, I end up sending them back a user with "is_deleted" field instead of domain-specific "isDeleted". 

Do you think @ArtiFisher that we need a mapping here back to "domain terms" (so we send `isDeleted` instead)?

Reply:
AFAIK data mapper is used to map to both directions. So yeah, it is preferable to do it, but I don't think that it's a blocker for a merge.